source/detail/parser/x86_64.cpp

```diff
@@ -364,13 +370,13 @@ namespace smeagle::x86_64 {
         st::Type *paramType = param->getType();
 
         // Get register class based on type
-        std::vector<RegisterClass> regClasses = getRegisterClassFromType(paramType);
+        std::pair<RegisterClass, RegisterClass> regClasses = getRegisterClassFromType(paramType);
 
         // Get the directionality (export or import) given the type
         std::string direction = getDirectionalityFromType(paramType);
 
         // Get register name from register classes
-        //      std::string loc = getStringLocationFromRegisterClass(regClasses);
+        std::string registers = getRegistersString(regClasses, order);
```

Comment:
The more I think about the problem of register allocation, the more I think we need a two-pass solution. For example, the ABI standard only refers to xmm0-xmm7. However, the AVX and AVX2 standards introduce ymm and zmm registers, respectively. These are not separate registers; the SSE registers alias subparts of the AVX and AVX2 register. Likewise, AVX aliases parts of the AVX2 registers. The standard doesn't explicitly state this as those are technically implementation details, but that's how both intel and AMD do it. Strangely, the example in standard _does_ show this aliasing.

The existing strategy will work fine for our simple `foo(int)`, but it'll need to be updated once we start looking at floating point types. Let's chat about this tomorrow, or when you have some time.

Reply:
I'll ping you tomorrow after my run and we can chat!